samples/CGCatalog/CGCatalog/Samples/CGCCGContextReplacePathWithStrokedPath.m

```diff
@@ -50,36 +57,86 @@ - (void)drawRect:(CGRect)rect {
+    CGContextSetLineCap(context, kCGLineCapRound);
+    CGContextSetLineJoin(context, kCGLineJoinMiter);
+    CGContextSetMiterLimit(context, 1);
+
+    if (self.strokedPathType == StrokedPathTypeReplaceStroke2) {
+        CGPathMoveToPoint(path, NULL, maxWidth / 2, maxHeight / 4);
+        CGPathAddLineToPoint(path, NULL, 16, 16);
+    } else {
+        CGPathMoveToPoint(path, NULL, 50, 50);
+        CGPathAddLineToPoint(path, NULL, 100, 100);
+    }
+
+    if (self.strokedPathType == StrokedPathTypeReplaceStroke2) {
+        CGPathMoveToPoint(path, NULL, maxWidth - 16, 16);
+        CGPathAddQuadCurveToPoint(path, NULL, maxWidth * 3 / 4, maxHeight * 1 / 4, maxWidth / 2, maxHeight * 3 / 4);
+        CGPathAddArcToPoint(path, NULL, maxWidth / 2, maxHeight * 1.25, 0, maxHeight / 4, 50);
+        CGPathAddLineToPoint(path, NULL, maxWidth / 4, maxHeight / 2);
+        CGPathAddLineToPoint(path, NULL, 10, maxHeight / 2);
+    } else {
+        CGPathMoveToPoint(path, NULL, 200, 200);
+        CGPathAddLineToPoint(path, NULL, 260, 100);
+        CGPathAddArcToPoint(path, NULL, 500, 500, 450, 450, 50);
+        CGPathAddQuadCurveToPoint(path, NULL, 100, 200, 200, 300);
+    }
+
     // Log original path info
     [self.logs addObject:@"Original Path:"];
     params = @{ @"logs" : self.logs, @"points" : originalPoints };
     CGPathApply(path, (__bridge void*)params, _Applier);
 
     // Create stroked path
     // TODO: Change this to use CGContextReplacePathWithStrokedPath()
-    NSMutableArray<NSValue*>* strokedPoints = [NSMutableArray new];
-    CGPathRef newPath = CGPathCreateCopyByStrokingPath(path, NULL, c_LineWidth, kCGLineCapButt, kCGLineJoinMiter, 1);
-    // Log stroked path info
-    [self.logs addObject:@"New Path:"];
-    params = @{ @"logs" : self.logs, @"points" : strokedPoints };
-    CGPathApply(newPath, (__bridge void*)params, _Applier);
 
     // Draw the path
     NSMutableArray<NSValue*>* points;
-    if (self.drawStrokedPath) {
-        CGContextSetFillColorWithColor(context, [UIColor blueColor].CGColor);
-        CGContextAddPath(context, newPath);
-        CGContextFillPath(context);
-        points = strokedPoints;
-    } else {
-        CGContextSetStrokeColorWithColor(context, [UIColor greenColor].CGColor);
-        CGContextAddPath(context, path);
-        CGContextStrokePath(context);
-        points = originalPoints;
+    switch (self.strokedPathType) {
+        case StrokedPathTypeLine: {
+            CGContextSetStrokeColorWithColor(context, [UIColor greenColor].CGColor);
+            CGContextAddPath(context, path);
+            CGContextStrokePath(context);
+            points = originalPoints;
+        } break;
+
+        case StrokedPathTypeCopyStroke: {
+            NSMutableArray<NSValue*>* strokedPoints = [NSMutableArray new];
+            CGPathRef newPath = CGPathCreateCopyByStrokingPath(path, NULL, c_LineWidth, kCGLineCapRound, kCGLineJoinMiter, 1);
```

Comment:
Yea, CGPathCreateCopyByStrokingPath isn't yet supported.   Added a NULL check and TODO to remove it once implemented.